packages/web3-eth-ens/src/contracts/Registry.js

```diff
@@ -63,14 +87,168 @@ Registry.prototype.owner = function (name, callback) {
                 promiEvent.resolve(receipt);
 
                 if (_.isFunction(callback)) {
-                    callback(receipt);
+                    // It's required to pass the receipt to the second argument to be backwards compatible and to have the required consistency
+                    callback(receipt, receipt);
+                }
+            })
+            .catch(function (error) {
+                promiEvent.reject(error);
+
+                if (_.isFunction(callback)) {
+                    callback(error, null);
+                }
+            });
+    });
+
+    return promiEvent.eventEmitter;
+};
+
+/**
+ * Returns the address of the owner of an ENS name.
+ *
+ * @method setOwner
+ *
+ * @param {string} name
+ * @param {string} address
+ * @param {TransactionConfig} txConfig
+ * @param {function} callback
+ *
+ * @callback callback callback(error, result)
+ * @returns {PromiEvent<TransactionReceipt | TransactionRevertInstructionError>}
+ */
+Registry.prototype.setOwner = function (name, address, txConfig, callback) {
+    var promiEvent = new PromiEvent(true);
+
+    this.contract.then(function (contract) {
```

Comment:
It's probably worth to abstract such contract method calls instead of duplicating a lot of code all the time. This could get addressed with a separate PR.